### ISSUE TITLE
wakeLockTime ..Does it kill the service after the timr is up?!

### ISSUE THREAD

**Issue by ahmadkhedr** [rekab-app/background_locator#130]
it is a QUESTION more than an issue...but does wakeLockTime stops the service automatically when the time mentioned is up?... Does it mean that the service will stop after 60 minutes if we left it as default?
I am using Xiaomi Redmi Note 7 to test my app and after some time the service stops and the sticky notification disappears..does this have something to do with this parameter or it's related to the device OS itself?

**Comment by mehdok** [comment]
No it doesn't kill the service after that time. `wakeLockTime` just guarantee that the phone will stay awake even when it's screen is off and prevent the phone from deep sleep;

About your problem with Xiaomi phones, we had a similar problem with Huawei phones and other Chinese Roms, They don't respect android laws and terminate the foreground service anytime they want. Sadly we don't have any solution for them yet;

**Comment by ahmadkhedr** [comment]
I have searched in the matter and found out that it really from the phone itself after as they say"whitelisted" my app on the device it worked the whole night without stopping the service ..I have stopped the background restriction and the battery optimization for my app manually from the settings.
I have found some links that can provide any help regarding solving this issue programmatically..did not try any of them yet and do not whether I will be able to apply them or not due to my small knowledge of the native code.
[https://stackoverflow.com/questions/59442078/background-services-are-restricted-in-xiaomi-and-oppo](url)
[https://stackoverflow.com/questions/48166206/how-to-start-power-manager-of-all-android-manufactures-to-enable-background-and](url)

**Comment by mehdok** [comment]
Thanks, I will investigate those links to see if we can address this issue programmatically;

**Comment by jeancarlos2020** [comment]
Añadí un permiso adicionar desde Java  y funciona 
https://developer.android.com/reference/android/Manifest.permission.html#REQUEST_IGNORE_BATTERY_OPTIMIZATIONS

**Comment by ferdinandfrank** [comment]
@mehdok One question regarding `wakeLockTime`: Is there any limit for the value of this option or any disadvantages for setting this value to a very high number, e.g., 1 day or 1 week?

**Comment by mehdok** [comment]
@ferdinandfrank 

Honestly I don't know any of that, because I'm not an Android expert.

Please take a look at [official doc](https://developer.android.com/training/scheduling/wakelock)

**Comment by RohitRB** [comment]
@mehdok were you successful in your investigation to fix this issue programatically?

**Comment by mehdok** [comment]
@RohitRB not really, I even merged some PR from others about these issue, but non of them fixed the problem.
